### PR TITLE
{lib}[GCCcore/13.2.0] OpenMPI v4.1.5, UCX v1.15.0-rc4, UCC v1.2.0, ...

### PR DESCRIPTION
**Pull request by SebastianAchilles**
(created using `eb --new-pr`)

Depends on:
- `pkgconf` https://github.com/easybuilders/easybuild-easyconfigs/pull/18706
- `Perl` and `Autotools` https://github.com/easybuilders/easybuild-easyconfigs/pull/18703
- `hwloc` https://github.com/easybuilders/easybuild-easyconfigs/pull/18707
- `libevent` https://github.com/easybuilders/easybuild-easyconfigs/pull/18708
- `libfabric` https://github.com/easybuilders/easybuild-easyconfigs/pull/18709

`UCX/1.14.1` does not compile with `GCC/13.2.0`, c.f. https://github.com/openucx/ucx/issues/9334
That is why`UCX/1.15.0-rc4` was used. Either we can merge this now and update to the final release later or we wait until the final version is released.

### CODE REVIEW

**Comment by SebastianAchilles** [comment]
Test report by @sebastianachilles
**SUCCESS**
Build succeeded for 5 out of 5 (5 easyconfigs in total)
bwd-rockylinux-92 - Linux Rocky Linux 9.2 (Blue Onyx), x86_64, Intel(R) Core(TM) i7-6900K CPU @ 3.20GHz (broadwell), 2 x NVIDIA NVIDIA GeForce GTX 1060 6GB, 535.54.03, Python 3.9.16
See https://gist.github.com/SebastianAchilles/4e0ed0acf300fb2f4a9b0ada5372dfeb for a full test report.

**Comment by branfosj** [comment]
@boegelbot please test @ generoso

**Comment by branfosj** [comment]
> That is whyUCX/1.15.0-rc4 was used. Either we can merge this now and update to the final release later or we wait until the final version is released.

I'm happy to merge this with the UCX RC, but I suggest we delay it until after the release of the next EB version.

**Comment by branfosj** [comment]
@boegelbot please test @ jsc-zen2

**Comment by branfosj** [comment]
Test report by @branfosj
**SUCCESS**
Build succeeded for 6 out of 6 (5 easyconfigs in total)
bear-pg0105u03a.bear.cluster - Linux RHEL 8.6, x86_64, Intel(R) Xeon(R) Platinum 8360Y CPU @ 2.40GHz (icelake), Python 3.6.8
See https://gist.github.com/branfosj/40fecc1c2be6cc21eae165d39664ca74 for a full test report.

**Review by branfosj — APPROVED** [review]

**Comment by branfosj** [comment]
Going in, thanks @SebastianAchilles!

File: easybuild/easyconfigs/u/UCX/UCX-1.15.0-rc4-GCCcore-13.2.0.eb [hunk 1]
```diff
@@ -0,0 +1,56 @@
+easyblock = 'ConfigureMake'
+
+name = 'UCX'
+version = '1.15.0-rc4'
```
Comment (boegel) [inline easybuild/easyconfigs/u/UCX/UCX-1.15.0-rc4-GCCcore-13.2.0.eb:4]: Are we OK with using a release candidate here? There's already an `rc6`, but we should probably stick to 1.14.1 instead?

File: easybuild/easyconfigs/g/gompi/gompi-2023b.eb [hunk 2]
```diff
@@ -0,0 +1,20 @@
+easyblock = 'Toolchain'
+
+name = 'gompi'
+version = '2023b'
```
Comment (boegel) [inline easybuild/easyconfigs/g/gompi/gompi-2023b.eb:4]: Note: this was changed to `2023.09` in #18888 for now, we don't consider `2023b` to be final just yet...

**Pull request merged**